source/common/http/utility.cc

```diff
@@ -478,6 +478,17 @@ Utility::parseHttp1Settings(const envoy::config::core::v3::Http1ProtocolOptions&
 
   if (config.header_key_format().has_proper_case_words()) {
     ret.header_key_format_ = Http1Settings::HeaderKeyFormat::ProperCase;
+  } else if (config.header_key_format().has_custom()) {
+    ret.header_key_format_ = Http1Settings::HeaderKeyFormat::Custom;
+    const auto& rules = config.header_key_format().custom().rules();
+    // Transform rules, inserting new elements into ret.header_key_format_rules map...
+    std::transform(rules.cbegin(), rules.cend(),
```

Comment:
There's actually a lower-case map implementation somewhere in this code base, as it turns out.